core/domain/classifier_services.py

```diff
@@ -127,6 +127,70 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+
+
+def add_to_training_queue(exploration):
+    """Checks all states of the exploration for re-training conditions and
+    creates ClassifierTrainingJobModel instances in the datastore
+    accordingly.
+
+    Args:
+        exploration: Exploration. Domain object for an exploration.
+    """
+    states = exploration.states
+    for state_name in states:
+        state = states[state_name]
+        if not state.can_undergo_classification():
+            continue
+        exp_id = exploration.id
+        exp_version = exploration.version
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        classifier = get_classifier_from_exploration_attributes(
+            exp_id, exp_version, state_name)
+        if classifier:
+            if not check_re_training_conditions(state, classifier.id):
+                create_classifier_exploration_mapping(exp_id, exp_version+1,
+                                                      state_name, classifier.id)
+                continue
+        create_classifier_training_job(algorithm_id, interaction_id, exp_id,
+                                       exp_version, state_name,
+                                       training_data,
+                                       feconf.TRAINING_JOB_STATUS_NEW)
+
+
+def check_re_training_conditions(state, job_id):
```

Comment:
Done. I changed It so is_retraining_necessary(). It should serve the purpose since its an internal function anyway.